demos-europe/demosplan-core
# Change build to support Tailwind in addons

This pull request addresses the issue of being unable to utilize Tailwind within addon markup. The current limitation stems from the fact that addons lack their own style build and heavily depend on styles generated in demosplan-core. Two potential solutions were considered:

1. Incorporate a styles build into each addon, supporting Tailwind with our custom configuration and including the compiled style in demosplan projects.
2. find a way to build demosplan-core styles after an addon is installed, integrating the addon paths into the demosplan-core styles build logic. This requires tailwindcss and purgeCss to be aware of these paths, ensuring classes used exclusively within addons are not inadvertently removed.

This pull request implements the second solution, which proved to be more feasible with minimal overhead. It introduces a new argument, `-S` (or `--only-styles`), to the frontend build command, enabling the exclusive building of styles for a faster process. Following addon installation, demosplan-core styles can now be built with this argument, incorporating the addon files into the tailwind config. As a result, Tailwind classes used in addons are included in the rendered styles.

### How to review/test
- Build the complete demosplan frontend with `./fe build <project>` 
- Add a tailwind class that is not present in demosplan-core in an addon file (in `addons/vendor/demosplan-europe/<addon>/client/`)
- Build the addon frontend (in `addons/vendor/demosplan-europe/<addon>/`) with `yarn prod`
- Build the demosplan frontend styles with `./fe build --only-styles <project>`

Then, navigate to the UI where your added class should be applied and make sure that the styles are present within the compiled css.

### PR Checklist

- [X] Update documentation in https://github.com/demos-europe/demosplan-documentation/pull/61
